pkg/service/controller/servingcert/secret_creating_controller.go

```diff
@@ -207,37 +232,48 @@ func (sc *ServiceServingCertController) work() bool {
 // syncService will sync the service with the given key.
 // This function is not meant to be invoked concurrently with the same key.
 func (sc *ServiceServingCertController) syncService(key string) error {
-	obj, exists, err := sc.serviceCache.GetByKey(key)
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
-		glog.V(4).Infof("Unable to retrieve service %v from store: %v", key, err)
 		return err
 	}
-	if !exists {
+	service, err := sc.serviceLister.Services(namespace).Get(name)
+	if kapierrors.IsNotFound(err) {
 		glog.V(4).Infof("Service has been deleted %v", key)
 		return nil
 	}
+	if err != nil {
+		return err
+	}
 
-	if !sc.requiresCertGeneration(obj.(*v1.Service)) {
+	secretUpdated, err := sc.updateSecretOwnerReference(service)
+	if err != nil {
+		return err
+	}
+	if secretUpdated {
```

Comment:
@deads2k wonder if we want to return here and reconcile with updated secret or just continue (my worry is the cache won't get the updated secret..)